src/modules/logs/__tests__/logs.effects.spec.ts

```diff
@@ -0,0 +1,185 @@
+    // ACT
+    actions$.next(new logsActions.SaveLog(log));
+    // ASSERT
+    effects.saveLogEffect$.subscribe((result) => {
+      expect(result instanceof logsActions.PersistLog).toBe(true);
+      done();
+    });
+  });
+
+  describe('persistLogs', () => {
+    it('should call saveLogs', fakeAsync((done) => {
+    // ARRANGE
+      spyOn(effects, 'saveLogs').and.callThrough();
+    // ACT
+      actions$.next(new logsActions.PersistLog());
+      tick();
+      // ASSERT
+      effects.persistLogEffect$.subscribe((result) => {
+        expect(effects.saveLogs).toHaveBeenCalled();
+        expect(result instanceof of).toBe(true);
+        done();
+      });
+    }));
+  });
+  describe('LoadLog', () => {
+    it('should call getPersistedLogs and return LoadLogState', fakeAsync((done) => {
+    // ARRANGE
+      spyOn(effects, 'getPersistedLogs').and.callThrough();
+    // ACT
+      actions$.next(new logsActions.LoadLog());
+      tick();
+      // ASSERT
+      effects.persistLogEffect$.subscribe((result) => {
+        expect(effects.getPersistedLogs).toHaveBeenCalled();
+        expect(result instanceof logsActions.LoadLogState).toBe(true);
+        done();
+      });
+    }));
+
+  });
+
+  describe('getAndConvertPersistedLogs', () => {
+    it('should empty cached data if cache is too old', () => {
+      const log: Log = { ['test']: 'xyz', type: LogType.DEBUG, message: 'test', timestamp: 1234567 };
+
+      const agedCache = {
+        dateStored: new DateTime().add(-(cacheDays + 1), Duration.DAY).format('YYYY/MM/DD'),
+        data: log,
+      };
+
+      // override mock getItem as we need data to test
+      // @ts-ignore
+      dataStoreMock.getItem.and.callFake(() => Promise.resolve(JSON.stringify(agedCache)));
+
+      spyOn(effects, 'emptyCachedData').and.callThrough();
+
+      effects.getAndConvertPersistedLogs().then((data) => {
+        expect(effects.emptyCachedData).toHaveBeenCalled();
+        expect(dataStoreMock.setItem).toHaveBeenCalled();
+        expect(data).toEqual([] as Log[]);
+      });
+    });
+
+    it('should return data without emptying cache if data is not too old', () => {
+      const log: Log = { ['test']: 'xyz', type: LogType.DEBUG, message: 'test', timestamp: 1234567 };
+      const dataWthinWindowCache = {
+        dateStored: new DateTime().add(cacheDays, Duration.DAY).format('YYYY/MM/DD'),
+        data: log,
+      };
+
+      // override mock getItem as we need data to test
+      // @ts-ignore
+      dataStoreMock.getItem.and.callFake(() => Promise.resolve(JSON.stringify(dataWthinWindowCache)));
+
+      spyOn(effects, 'emptyCachedData').and.callThrough();
+
+      effects.getAndConvertPersistedLogs().then((data) => {
+        expect(effects.emptyCachedData).toHaveBeenCalledTimes(0);
+        expect(dataStoreMock.setItem).toHaveBeenCalledTimes(0);
+        expect(data).toEqual(dataWthinWindowCache.data);
```

Comment:
Just to be safe, could we please add some `done()` calls here at the end of these `.then` calls.

Reply:
done